### PR TITLE
let service handle GUI log writes and rotations

### PR DESCRIPTION
**Pull request by songgao**

### CODE REVIEW

File: go/status/log_send.go [hunk 1]
```diff
@@ -59,6 +59,7 @@ type LogSendContext struct {
 	kbfsLog          string
 	svcLog           string
 	ekLog            string
+	guiLog           string
```
Comment (joshblum) [inline go/status/log_send.go]: why do we need both `guiLog` and `desktopLog`?

File: go/libkb/constants.go [hunk 2]
```diff
@@ -576,10 +576,11 @@ const (
 const (
 	ServiceLogFileName = "keybase.service.log"
 	EKLogFileName      = "keybase.ek.log"
+	GUILogFileName     = "keybase.gui.log"
 	KBFSLogFileName    = kbconst.KBFSLogFileName
 	GitLogFileName     = "keybase.git.log"
 	UpdaterLogFileName = "keybase.updater.log"
-	DesktopLogFileName = "Keybase.app.log"
+	DesktopLogFileName = "Keybase.app.log" // deprecated; use GUILogFileName
```
Comment (joshblum) [inline go/libkb/constants.go]: why not use keep the same filename?

Comment (songgao) [reply go/libkb/constants.go]: I was afraid I'd mess it up. But I guess it's not a huge deal to lose some logs here .. will update to use the same name.

Comment (songgao) [reply go/libkb/constants.go]: actually I do want to lowercase that. So to avoid confusing maybe it's best to keep the `.gui` part?

Comment (songgao) [reply go/libkb/constants.go]: ended up keeping `Keybase.app.log`

**Comment by songgao** [comment]
@joshblum Thanks! ready for another look

File: go/bind/keybase.go [hunk 3]
```diff
@@ -285,7 +285,11 @@ func LogSend(statusJSON string, feedback string, sendLogs, sendMaxBytes bool, ui
 	logSendContext.InstallID = env.GetInstallID()
 	logSendContext.StatusJSON = statusJSON
 	logSendContext.Feedback = feedback
-	logSendContext.Logs.Desktop = uiLogPath
+	if len(uiLogPath) == 0 {
```
Comment (joshblum) [inline go/bind/keybase.go]: this should be `!=` right? but i think we should just drop this param entirely and let the service run the show

Comment (songgao) [reply go/bind/keybase.go]: oops yes. cool will do!

**Comment by songgao** [comment]
OK ready for another look @joshblum thanks! There was actually an issue with Android; fixed by using the passed-in path like the ekLog.

**Comment by songgao** [comment]
@chrisnojima Yo! This still need a review on JS side. Mind taking a look when you have time? Thanks!

**Pull request merged**